packages/next-plugin-sentry/src/on-error-server.js

```diff
@@ -0,0 +1,36 @@
+import { captureException, flush, Handlers, withScope } from '@sentry/nextjs';
+import getConfig from 'next/config';
+
+const { parseRequest } = Handlers;
+
+export default async function onErrorServer(err) {
+  const { serverRuntimeConfig = {}, publicRuntimeConfig = {} } = getConfig() || {};
+  const sentryTimeout = serverRuntimeConfig.sentryTimeout || publicRuntimeConfig.sentryTimeout || 2000;
+
+  withScope(scope => {
+    if (typeof err.req !== 'undefined') {
+      scope.addEventProcessor(event => {
+        return parseRequest(event, err.req, {
+          // 'cookies' and 'query_string' use `dynamicRequire` which has a bug in SSR envs right now — Kamil
+          request: ['data', 'headers', 'method', 'url'],
+        });
+      });
+    }
+
+    const toCapture = err instanceof Error ? err : err.err;
+
+    scope.addEventProcessor((event, hint) => {
+      if (hint.originalException === toCapture) {
+        event.exception.values[0].mechanism = {
+          handled: false,
+          type: 'onErrorServer',
+        };
+      }
+      return event;
+    });
+
+    captureException(toCapture);
+  });
+
+  await flush(sentryTimeout);
```

Comment:
This is probably fine, but has a caveat: it possibly slows down requests on the server when there are errors, even if it is not necessary.

Flushing on every error is most of the time not required. In certain platforms, in particular serverless platforms, we should flush once per request.

Is there an obvious better place we could put it in? Per-request flush?